spec/graphql/query/executor_spec.rb

```diff
@@ -132,6 +132,22 @@
       end
     end
 
+    describe 'if nil is given for a non-null field' do
+      let(:query_string) {%| query noMilk { cantBeNullButIs }|}
+      it 'turns into error messages' do
+        expected = {
+          "data" => { "cantBeNullButIs" => nil },
+          "errors" => [
+            {
+              "message" => "Type mismatch resolving 'cantBeNullButIs'",
+              "locations" => [ { "line" => 1, "column" => 17 } ]
+            }
+          ]
+        }
+        assert_equal(expected, result)
+      end
+    end
+
```

Comment:
Could you add another test for a field nested within an object to make sure only one error appears in the errors list?